gcsio/src/main/java/com/google/cloud/hadoop/gcsio/GoogleCloudStorageImpl.java

```diff
@@ -267,24 +267,11 @@ public GoogleCloudStorageImpl(
    *
    * @param gcs Preconstructed Storage to use for I/O.
    */
-  public GoogleCloudStorageImpl(GoogleCloudStorageOptions options, Storage gcs) {
-    Preconditions.checkNotNull(options, "options must not be null");
-
-    logger.atFine().log("GCS(%s)", options.getAppName());
-
-    options.throwIfNotValid();
-
-    this.storageOptions = options;
-
+  public GoogleCloudStorageImpl(GoogleCloudStorageOptions options, Storage gcs) throws IOException {
```

Comment:
I think that it should be implemented differently - we should reuse prod constructor in test constructor, not opposite.

Reply:
Are both of these public API that can be used in prod?

Reply:
No, one of them has `@VisibleForTesting` annotation which means that it intended to be used in tests.